ray-operator/apis/ray/v1/raycluster_types.go

```diff
@@ -167,8 +167,13 @@ type RayClusterStatus struct {
 type RayClusterConditionType string
 
 const (
-	// HeadReady is added in a RayCluster when its Head Pod is ready for requests.
-	HeadReady RayClusterConditionType = "HeadReady"
+	// PodRunningAndReady says that the pod is running and ready.
+	PodRunningAndReady = "PodRunningAndReady"
+)
+
+const (
+	// HeadPodReady is added in a RayCluster when its Head Pod is ready for requests.
+	HeadPodReady RayClusterConditionType = "Ready"
```

Comment:
```suggestion
	HeadPodReady RayClusterConditionType = "HeadPodReady"
```

Reply:
Use `HeadPodReady` instead. The RayCluster's "Ready" status is not equivalent to the definition of "Ready" for the head pod.

* `HeadPodReady`: The head Pod is ready. This means that the GCS, Ray dashboard, and Raylet are ready to serve requests. Hence, the KubeRay operator can send requests like `/api/jobs/` to the Ray dashboard to query Ray job status or create a Ray job. However, it doesn't promise that the cluster is ready to execute the job.

* RayCluster's "Ready": The RayCluster is ready to create and execute a Ray job. This includes that:
  * The head Pod is ready, so KubeRay can send requests to the Ray dashboard.
  * The cluster has enough worker pods up and running.